### ISSUE TITLE
How to add a dictionary ?

### ISSUE THREAD

**Issue by Yoannjap** [tuberry/light-dict#1]
Hello,

You have just the most brillant idea ever of a possible gnome extension. Thanks a lot for that?

However, I haven't understood the way to enter a search reference in the Advanced options.

For example, how to add an icon shortcut to search a Wordref translation from English to French like in https://www.wordreference.com/enfr/ ?

Or to add a French definition like with this URL : https://www.lalanguefrancaise.com/dictionnaire/?s=
?

Thank you for your help.

And bravo for your Light-Dict extension.

Yoannjap

**Comment by tuberry** [comment]
Thanks, the extension is in developing and I haven't finalized the design.
As for icons, I usually add them to `~/.local/share/icons/hicolor/symbolic/apps/`;
I am not sure what you mean. 
If you want to search word with that URL, that's easy. Just replace the word with `LDWORD` and open the new URL with `gio open`:
`search#000#gio open https://www.wordreference.com/enfr/LDWORD`
![image](https://user-images.githubusercontent.com/17917040/79772000-20e2f580-8362-11ea-9890-3ded6716b3eb.png)
And you will see icon shortcut after selecting any text:
![image](https://user-images.githubusercontent.com/17917040/79772655-e88fe700-8362-11ea-84b0-dfcb3bb79da8.png)
Or you want to show the definition in popup panel, that means more works for you:         

1.  write or find a web crawler to get the result;
2.  add the script to your local path so that you can run it in Bash like `translate word`;
3.  add an entry in Advanced: `search#100#translate LDWORD`

In short,  a command-line version dictionary is required. You might like [dictd](https://wiki.archlinux.org/index.php/Dictd) or [translate-shell](https://github.com/soimort/translate-shell).

**Comment by Yoannjap** [comment]
Thanks for your reply. Very complete.

What I would like is when I highlight the word, is to get 3 icons appearing above the word, one redirecticting in a popup to https://www.lalanguefrancaise.com/dictionnaire/WORD if I click on it ; a second one redirecting in a pop up to https://www.wordreference.com/fres/WORD if I click on it.

That's exactly what could be great with your addon.

**Comment by tuberry** [comment]
What this extension does is just excuting the command you given and showing the result in the popup  panel if necessary. So fetching and parsing the Web content is beyond its extent. If you just need shortcuts to open URL in browser, there is no difference between 1 icon and 3 icons. They can be added in the mostly same way.

**Comment by tuberry** [comment]
I will close this if there is no more problem.
